### PR TITLE
Nd pooling gpu

### PR DESCRIPTION
**Pull request by kahmed10**

### CODE REVIEW

**Comment by pfultz2** [comment]
You need to add `pkg-config` to the list of packages in the Dockerfile.

**Comment by mvermeulen** [comment]
Not sure what is going on with Jenkins.  However, as an experiment, I tried the following build with ROCm 3.5 including MIOpen 2.4:

1. Run docker container mevermeulen/rocm-migraphx:20200629
2. cd /src/AMDMIGraphX
3. git pull
4. git checkout	nd_pooling_gpu
5. mkdir build_nd_pooling_gpu
6. cd build_nd_pooling_gpu
7. env CXX=/opt/rocm/llvm/bin/clang++ cmake ..
8. make
9. make	check

This successfully built and also passed all the tests.

File: src/include/migraphx/op/pooling.hpp [hunk 1]
```diff
@@ -38,13 +38,18 @@ struct pooling
...
         {
-            MIGRAPHX_THROW("pooling: inconsistent attribute sizes");
+            MIGRAPHX_THROW("convolution: inconsistent attribute sizes");
         }
+    }
+
+    shape compute_shape(std::vector<shape> inputs) const
+    {
+        check_shapes{inputs, *this}.has(1);
+        check_attribute_size();
```
[the remark this answers is not in the document]
Comment (kahmed10) [reply src/include/migraphx/op/pooling.hpp]: changed

File: src/include/migraphx/op/pooling.hpp [hunk 2]
```diff
@@ -38,13 +38,18 @@ struct pooling
...
     std::string name() const { return "pooling"; }
 
-    shape compute_shape(std::vector<shape> inputs) const
+    void check_attribute_size() const
     {
-        check_shapes{inputs, *this}.has(1);
         if(not(padding.size() == stride.size() and padding.size() == lengths.size()))
         {
-            MIGRAPHX_THROW("pooling: inconsistent attribute sizes");
+            MIGRAPHX_THROW("convolution: inconsistent attribute sizes");
```
[the remark this answers is not in the document]
Comment (kahmed10) [reply src/include/migraphx/op/pooling.hpp]: changed

File: src/include/migraphx/op/pooling.hpp [hunk 3]
```diff
@@ -38,19 +38,27 @@ struct pooling
...
-        check_shapes{inputs, *this}.has(1);
         if(not(padding.size() == stride.size() and padding.size() == lengths.size()))
         {
             MIGRAPHX_THROW("pooling: inconsistent attribute sizes");
         }
+    }
+
+    shape compute_shape(std::vector<shape> inputs) const
+    {
+        check_shapes{inputs, *this}.has(1);
```
Comment (scxiao) [inline src/include/migraphx/op/pooling.hpp:52]: Looks like we missed the function call check_attribute_size() here

Comment (kahmed10) [reply src/include/migraphx/op/pooling.hpp:52]: it is implicitly called when doing this->kdims() in L58

**Pull request merged**